src/hooks/OnIncompatibleSchemaChange/useSpecificationIncompatibleSchemaSetting.ts

```diff
@@ -0,0 +1,69 @@
+import { modifyDraftSpec } from 'api/draftSpecs';
+import {
+    useEditorStore_persistedDraftId,
+    useEditorStore_queryResponse_draftSpecs,
+    useEditorStore_queryResponse_mutate,
+} from 'components/editor/Store/hooks';
+import { AutoCompleteOption } from 'components/incompatibleSchemaChange/types';
+import { omit } from 'lodash';
+import { useCallback, useMemo } from 'react';
+import { logRocketEvent } from 'services/shared';
+import { CustomEvents } from 'services/types';
+import { Schema } from 'types';
+import { hasLength } from 'utils/misc-utils';
+
+export default function useSpecificationIncompatibleSchemaSetting() {
+    // Draft Editor Store
+    const draftId = useEditorStore_persistedDraftId();
+    const draftSpecs = useEditorStore_queryResponse_draftSpecs();
+    const mutateDraftSpecs = useEditorStore_queryResponse_mutate();
+
+    const draftSpec = useMemo(
+        () =>
+            draftSpecs.length > 0 && draftSpecs[0].spec ? draftSpecs[0] : null,
+        [draftSpecs]
+    );
+
+    const updateOnIncompatibleSchemaChange = useCallback(
+        async (value: AutoCompleteOption['val'] | undefined) => {
+            if (!mutateDraftSpecs || !draftId || !draftSpec) {
+                logRocketEvent(
+                    `${CustomEvents.INCOMPATIBLE_SCHEMA_CHANGE} : missing critical resources to update draft`,
```

Comment:
Don't write sentences in the eventing. These should be in between human and machine readable... like `${CustomEvents.INCOMPATIBLE_SCHEMA_CHANGE}:Missing Fields`.

Reply:
This was an oversight on my end as it was a console log originally. Will update.